### PR TITLE
 [Idea]: Buchberger's Algorithm #1514 

### LINKED ISSUE #1514: [Idea]: Buchberger's Algorithm
### Idea Title

Buchberger's Algorithm implementation 

### Idea Description

I will add Buchberger's Algorithm implementation, which will help users gain knowledge about this algo too.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

Assign me this under hacktoberfest and gssoc-ext

### PR DESCRIPTION
**Pull request by AE-Hertz**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

 [Idea]: Buchberger's Algorithm #1514 

Fixes #1514 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/Buchberger's Algorithm/Buchberger's Algorithm.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+- **Algorithm Type**: Symbolic computation, Gröbner Basis computation.
+- **Main Operation**: Computes the S-polynomial of polynomial pairs and reduces them.
+- **Data Structures**: A vector of polynomials is maintained during the computation.
+- **Output**: A Gröbner Basis for the input polynomials.
+
+### Time Complexity
+
+- **Average Case**: The complexity depends on the degree of the polynomials and the number of terms. The algorithm can have exponential time complexity in the worst case.
+- **Worst Case**: \( O(n^3) \) or more, depending on the implementation and the number of polynomials involved.
+
```
Comment (ajay-dhangar) [inline docs/algorithms/Buchberger's Algorithm/Buchberger's Algorithm.md]: replace `\( O(n^3) \)` with `$ O(n^3) $`

File: docs/algorithms/Buchberger's Algorithm/Buchberger's Algorithm.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+### Space Complexity
+
+- **Space Complexity**: \( O(n) \), where \( n \) is the number of polynomials. The space is required to store the polynomials and their intermediate results.
```
Comment (ajay-dhangar) [inline docs/algorithms/Buchberger's Algorithm/Buchberger's Algorithm.md]: use `$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by AE-Hertz** [comment]
> update your code based on suggestions

resolved/

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**